libgnucash/engine/Account.cpp

```diff
@@ -6226,6 +6227,39 @@ gboolean xaccAccountRegister (void)
     return qof_object_register (&account_object_def);
 }
 
+
+using AccountSet = std::unordered_set<Account*>;
+static void add_descendant (Account *acc, gpointer data)
+{
+    static_cast<AccountSet*> (data)->emplace (acc);
```

Comment:
`insert`, not `emplace`: `emplace` only works for C++ objects with constructors and expects a parameter list; `insert` expects an object reference and that's what you're passing.